### PR TITLE
Upgraded to java 11

### PR DESCRIPTION
**Pull request by AzGoalie**
Upgrade ddf-ui to java 11.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -45,19 +45,26 @@
   <properties>
     <!--Project properties-->
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
-    <maven.compiler.source>1.8</maven.compiler.source>
-    <maven.compiler.target>1.8</maven.compiler.target>
+    <maven.compiler.source>11</maven.compiler.source>
+    <maven.compiler.target>11</maven.compiler.target>
     <compiler.fork>false</compiler.fork>
-    <maven.deploy.plugin.version>2.8.2</maven.deploy.plugin.version>
+    <maven.bundle.plugin.version>4.2.1</maven.bundle.plugin.version>
```
Comment (AzGoalie) [inline pom.xml:51]: Upgraded a few maven plugins too

File: pom.xml [hunk 2]
```diff
@@ -45,19 +45,26 @@
...
-    <errorprone.version>2.2.0</errorprone.version>
+    <maven-jar-plugin.version>3.2.0</maven-jar-plugin.version>
+    <errorprone.version>2.4.0</errorprone.version>
+    <jacoco.maven.plugin.version>0.8.6</jacoco.maven.plugin.version>
+    <fmt.maven.plugin.version>2.3.0</fmt.maven.plugin.version>
 
     <!--Backend properties-->
-    <ddf.version>2.26.5</ddf.version>
-    <ddf-jsonrpc.version>0.6</ddf-jsonrpc.version>
+    <ddf.version>2.27.0-SNAPSHOT</ddf.version>
```
Comment (AzGoalie) [inline pom.xml]: java 11 ddf hasn't been released yet

File: pom.xml [hunk 3]
```diff
@@ -88,6 +95,7 @@
     <pac4j.version>4.0.1</pac4j.version>
     <pax.web.jsp.version>7.2.18</pax.web.jsp.version>
     <quartz.version>2.3.2</quartz.version>
+    <restassured.version>3.3.0</restassured.version>
```
Comment (AzGoalie) [inline pom.xml:98]: using restassured's own maven artifact now and not relying on ddf-thirdparty

File: pom.xml [hunk 4]
```diff
@@ -168,42 +176,37 @@
...
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-resources-plugin</artifactId>
-          <version>3.0.2</version>
+          <version>${maven.resources.plugin.version}</version>
         </plugin>
         <plugin>
           <groupId>org.apache.maven.plugins</groupId>
           <artifactId>maven-compiler-plugin</artifactId>
-          <version>3.8.1</version>
+          <version>${maven.compiler.plugin.version}</version>
```
Comment (AzGoalie) [inline pom.xml:184]: These settings are the same as ddf

File: ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/model/FilterNodeMapImpl.java [hunk 5]
```diff
@@ -67,12 +67,9 @@ public String getOperator() {
 
   @Override
   public List<FilterNode> getChildren() {
-    return Stream.of(json.get(CHILDREN))
-        .map(List.class::cast)
-        .flatMap(List::stream)
-        .map(Map.class::cast)
-        .map(FilterNodeMapImpl::new)
-        .collect(Collectors.toList());
+    List<Object> filters = (List<Object>) json.get(CHILDREN);
```
Comment (AzGoalie) [inline ui-backend/catalog-ui-search/src/main/java/org/codice/ddf/catalog/ui/forms/model/FilterNodeMapImpl.java:70]: java 11 doesn't like the type errasure that the previous version had. Made it more explicit.

File: ui-backend/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandlerTest.java [hunk 6]
```diff
@@ -86,7 +86,7 @@
   private static final String SERVICE_NOT_FOUND = "\"Service not found\"";
   private static final String SERVICE_SUCCESS = GSON.toJson("");
   private static final String ATTACHMENT_REGEX =
-      "^attachment;filename=\"export-\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}(\\.\\d{3})?Z."
+      "^attachment;filename=\"export-\\d{4}-\\d{2}-\\d{2}T\\d{2}:\\d{2}:\\d{2}(\\.(\\d{3}){1,2})?Z."
```
Comment (AzGoalie) [inline ui-backend/catalog-ui-search/src/test/java/org/codice/ddf/catalog/ui/query/handlers/CqlTransformHandlerTest.java:89]: Somewhere between java 8 and 11 the default time instance became more precise.

**Comment by AzGoalie** [comment]
build now

File: pom.xml (hunk 2 above)
Comment (shaundmorris) [reply pom.xml]: That's a good point, we probably should plan on a DDF release sometime soon. Maybe after a few more upgrades so we can release the whole DDF stack

**Review by shaundmorris — APPROVED** [review]

**Review by stevenmalmgren — APPROVED** [review]

**Review by brianfelix — APPROVED** [review]

**Review by mazarag2 — APPROVED** [review]

**Review by vinamartin — APPROVED** [review]

**Comment by TonyMorrison** [comment]
build now

**Comment by abel-connexta** [comment]
:tada: Hero Successful :tada:
Tested simple and intrigue ui basic functionalities and queried ingested data, and also checked admin-ui configuration and basic functionalities as well and all working as intended.
LGTM!

**Comment by shaundmorris** [comment]
The CI build has passed on Jenkins directly with a replay using JDK 11. So this is good to merged. We'll merge the JDK 11 Jenkins file change on this PR: https://github.com/codice/ddf-ui/pull/577

**Pull request merged**